### PR TITLE
[RW-982][RW-987][RW-984][risk=no] API calls for meds search

### PR DESCRIPTION
**Pull request by freemabd1**
We have 4 api calls for meds search
1) call to load the med ingredients tree
2) call to find autocomplete results in search box
3) call to find ingredients if a brand is selected
4) call to get participant count for the meds selected

### CODE REVIEW

File: api/src/main/resources/cb_search_api.yaml [hunk 1]
```diff
@@ -110,6 +110,47 @@ paths:
...
             $ref: "#/definitions/CriteriaListResponse"
 
+  /v1/cohortbuilder/{cdrVersionId}/criteria/drug/{name}:
+    parameters:
+      - $ref: '#/parameters/cdrVersionId'
+    get:
+      tags:
+        - cohortBuilder
+      description: Returns criteria tree with the specified name
+      operationId: "getDrugBrandOrIngrediantByName"
```
Comment (dolbeew) [inline api/src/main/resources/cb_search_api.yaml]: nit: change 'Ingrediant' to 'Ingredient' to avoid possible future issues

Comment (freemabd1) [reply api/src/main/resources/cb_search_api.yaml]: Nice catch. Done.

File: api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java [hunk 2]
```diff
@@ -112,6 +112,28 @@
     return ResponseEntity.ok(criteriaResponse);
   }
 
+  @Override
+  public ResponseEntity<CriteriaListResponse> getDrugBrandOrIngrediantByName(Long cdrVersionId, String drugName) {
```
[the remark this answers is not in the document]
Comment (freemabd1) [reply api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java]: Doh. Nice catch. done.

File: api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java [hunk 3]
```diff
@@ -9,43 +9,59 @@
...
+    "order by c.name asc", nativeQuery = true)
+  List<Criteria> findCriteriaByTypeAndSubtypeOrderByNameAsc(@Param("type") String type, @Param("subtype") String subtype);
+
+  @Query(value = "select * from criteria c " +
+    "where c.type = 'DRUG' " +
+    "and c.subtype in ('ATC', 'BRAND') " +
+    "and c.is_group = 0 and c.is_selectable = 1 " +
+    "and upper(c.name) like %:name% " +
+    "order by c.name asc", nativeQuery = true)
+  List<Criteria> findDrugBrandOrIngrediantByName(@Param("name") String name);
```
[the remark this answers is not in the document]
Comment (freemabd1) [reply api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java]: done.

File: api/src/main/resources/cb_search_api.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (freemabd1) [reply api/src/main/resources/cb_search_api.yaml]: done.

File: api/src/test/java/org/pmiops/workbench/cdr/dao/CriteriaDaoTest.java [hunk 4]
```diff
@@ -143,6 +169,28 @@ public void findCriteriaByTypeAndSubtypeOrderByNameAsc() throws Exception {
     assertEquals(demoCriteria1a, demoList.get(1));
   }
 
+  @Test
+  public void findDrugBrandOrIngrediantByName() throws Exception {
```
[the remark this answers is not in the document]
Comment (freemabd1) [reply api/src/test/java/org/pmiops/workbench/cdr/dao/CriteriaDaoTest.java]: done.

File: api/src/test/java/org/pmiops/workbench/cdr/dao/CriteriaDaoTest.java [hunk 5]
```diff
@@ -143,6 +169,28 @@ public void findCriteriaByTypeAndSubtypeOrderByNameAsc() throws Exception {
...
+    List<Criteria> drugList = criteriaDao.findDrugBrandOrIngrediantByName("ETAM");
+    assertEquals(1, drugList.size());
+    assertEquals(drugCriteriaIngredient, drugList.get(0));
+
+    drugList = criteriaDao.findDrugBrandOrIngrediantByName("ACE");
+    assertEquals(1, drugList.size());
+    assertEquals(drugCriteriaIngredient, drugList.get(0));
+
+    drugList = criteriaDao.findDrugBrandOrIngrediantByName("BL");
```
[the remark this answers is not in the document]
Comment (freemabd1) [reply api/src/test/java/org/pmiops/workbench/cdr/dao/CriteriaDaoTest.java]: done.

**Pull request merged**